x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -242,15 +269,113 @@ export const DetailsPageMappingsContent: FunctionComponent<{
+                  color="success"
+                  fill
+                  disabled={newFieldsLength === 0 ?? true}
+                  data-test-subj="indexDetailsMappingsSaveMappings"
+                >
+                  <FormattedMessage
+                    id="xpack.idxMgmt.indexDetails.mappings.saveMappings"
+                    defaultMessage="Save mappings"
+                  />
+                </EuiButton>
+              )}
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                data-test-subj="indexDetailsMappingsToggleViewButton"
+                onClick={onToggleChange}
+              >
+                {isJSONVisible ? (
+                  <FormattedMessage
+                    id="xpack.idxMgmt.indexDetails.mappings.tableView"
+                    defaultMessage="List"
+                  />
+                ) : (
+                  <FormattedMessage
+                    id="xpack.idxMgmt.indexDetails.mappings.json"
+                    defaultMessage="JSON"
+                  />
+                )}
+              </EuiButton>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          {addFieldComponent && (
+            <EuiFlexItem grow={false}>
+              <EuiAccordion
+                id={pendingFieldListId}
+                initialIsOpen
+                buttonContent={
+                  <EuiPanel paddingSize="s" hasShadow={false}>
+                    <EuiFlexGroup
+                      gutterSize="s"
+                      alignItems="baseline"
+                      data-test-subj="indexDetailsMappingsPendingBlock"
+                    >
+                      <EuiFlexItem grow={6}>
+                        <EuiText size="m">
+                          <FormattedMessage
+                            id="xpack.idxMgmt.indexDetails.mappings.addMappingPendingBlock"
+                            defaultMessage="Pending fields"
+                          />
+                        </EuiText>
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiNotificationBadge
+                          color={newFieldsLength > 0 ? 'accent' : 'subdued'}
+                          size="m"
+                        >
+                          {newFieldsLength}
+                        </EuiNotificationBadge>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  </EuiPanel>
+                }
+                borders="all"
+              >
+                <EuiPanel>
+                  <DocumentFields />
+                </EuiPanel>
+              </EuiAccordion>
+            </EuiFlexItem>
+          )}
 
-        <EuiFlexItem
-          grow={3}
-          css={css`
-            min-width: 600px;
-          `}
-        >
-          <EuiPanel>{isJSONVisible ? jsonBlock : treeViewBlock}</EuiPanel>
-        </EuiFlexItem>
+          <EuiFlexItem
+            grow={false}
```

Comment:
this might be an unintended change, I think we need to keep `grow={3}` for the vertical split between the main content and docs link

Reply:
Added this to remove vertical space between space bar and json/list view content when there are no mappings. 

With `grow={3}` 

<img width="1711" alt="Screenshot 2024-03-25 at 10 47 01 AM" src="https://github.com/elastic/kibana/assets/55930906/d0fdbd1f-2048-4711-9628-ab8a538b5787">


With `grow={false}`

<img width="1711" alt="Screenshot 2024-03-25 at 10 47 27 AM" src="https://github.com/elastic/kibana/assets/55930906/0baa6557-d229-4782-ab1b-58ca7b62d38a">

Reply:
yes, I see the issue, the new content should go inside the flex item with grow=3. Those are like the 2 main blocks in the tab, one for main content on the left (grow=3) and the other one for doc links (grow=1). I think it's okay to merge this code as is, we need to add a layout component for that to enforce the 3-to-1 split, I'll open an issue